aiven/resource_vpc_peering_connection.go

```diff
@@ -286,33 +287,31 @@ func resourceVPCPeeringConnectionRead(_ context.Context, d *schema.ResourceData,
 	return copyVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
 }
 
-func resourceVPCPeeringConnectionDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*aiven.Client)
 
 	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
-	isAzure, err := isAzureVPCPeeringConnection(d, client)
-	if err != nil {
-		return diag.Errorf("Error checking if it Azure VPC peering connection: %s", err)
-	}
 
-	if isAzure {
-		if peerResourceGroup, ok := d.GetOk("peer_resource_group"); ok {
-			err := client.VPCPeeringConnections.DeleteVPCPeeringWithResourceGroup(
-				projectName, vpcID, peerCloudAccount, peerVPC, peerResourceGroup.(string), peerRegion)
-			if err != nil {
-				return diag.Errorf("Error deleting VPC peering connection with resource group: %s", err)
-			}
-			return nil
-		} else {
-			return diag.Errorf("cannot delete an Azure VPC peering connection without `peer_resource_group`")
-		}
+	var peerResourceGroupPtr *string
```

Comment:
Hi, VPC peering connection may be in one of these states:
- ACTIVE
- REJECTED_BY_PEER 
- PENDING_PEER
- INVALID_SPECIFICATION
- DELETING
- DELETED
- DELETED_BY_PEER
- APPROVED

When we initiate deletion, it should switch to the **DELETING** state, and after a while, should change to **DELETED**, and then be gone from the system.

Let's rework the deletion waiter slightly. First of all, I would recommend keeping the existing deletion logic, and in the deletion waiter, we can query the VPC peering connection. If we get back the `404 NotFound` error, we consider the status to be **DELETED**. Otherwise, we take what is coming from the API.

Something like that:
```golang
func (w *VPCPeeringConnectionDeleteWaiter) RefreshFunc() resource.StateRefreshFunc {
	return func() (interface{}, string, error) {
		pc, err := w.Client.VPCPeeringConnections.GetVPCPeering(w.Project, w.VPCID, w.PeerCloudAccount, w.PeerVPC, w.PeerRegion)

		if err != nil {
			if aiven.IsNotFound(err) {
				return nil, "DELETED", nil
			}
			return nil, "", err
		}

		return pc, pc.State, nil
	}
}
```

Moreover, I was checking whether we can get away with checking `peer_resource_group` and determining whether it is Azure or not. I think we should still use `isAzureVPCPeeringConnection` to determine that because the user by accident may set this optional field but still VPC will be created in AWS or GCP cloud, so we should be sure that we are deleting an Azure VPC.